src/namespace.js

```diff
@@ -0,0 +1,81 @@
+
+export const namespaceSeparator = '/'
+
+export function validateNamespace (namespace) {
+  if (typeof namespace === 'string') {
+    namespace = namespace.split(namespaceSeparator)
+  }
+
+  if (!Array.isArray(namespace)) {
+    // TODO: warn?
+    return []
+  }
+
+  return namespace.filter(Boolean)
+}
+
+export function getPropertyByNamespace (object, namespace) {
+  namespace = validateNamespace(namespace)
+
+  for (const name of namespace) {
+    if (name in object) {
+      object = object[name]
+      continue
+    }
+
+    // TODO: warn on else?
+  }
+
+  return object
+}
+
+export function setPropertyByNamespace (object, namespace, value) {
+  namespace = validateNamespace(namespace)
+
+  const key = namespace.pop()
+
+  for (const name of namespace) {
+    if (name in object) {
+      object = object[name]
+      continue
+    }
+
+    // TODO: warn on else?
+  }
+
+  object[key] = value
```

Comment:
Maybe use Vue.set? Not sure if this is supposed to work when a prop doesnt exist yet